docs/docs/testing-components-with-graphql.md

```diff
@@ -30,39 +30,46 @@ done.
 As this is testing a page component you will need to put your tests in another
 folder so that Gatsby doesn't try to turn the tests into pages.
 
-```js:title=src/__tests__/index.js
+```js:title=src/pages/__tests__/index.js
 import React from "react"
 import renderer from "react-test-renderer"
-import BlogIndex from "../pages/index"
+import Index from "../index"
 
-describe("BlogIndex", () =>
+describe("Index", () =>
   it("renders correctly", () => {
-    const tree = renderer.create(<BlogIndex />).toJSON()
+    const tree = renderer.create(<Index />).toJSON()
     expect(tree).toMatchSnapshot()
   }))
 ```
 
-If you run this test you will get an error, as the component is expecting a
-location object. You can fix this by passing one in:
+If you run this test you will get an error, as the StaticQuery in the `Layout` component is not mocked. You can fix this by mocking it, like so:
 
 ```js:title=src/__tests__/index.js
 import React from "react"
 import renderer from "react-test-renderer"
-import BlogIndex from "../pages/index"
+import { StaticQuery } from "gatsby"
+import Index from "../pages/index"
 
-describe("BlogIndex", () =>
-  it("renders correctly", () => {
-    const location = {
-      pathname: "",
-    }
+beforeEach(() => {
+  StaticQuery.mockImplementationOnce(({ render }) =>
```

Comment:
We should think about how we can improve this 🤔 It's very not beginner friendly.